### PR TITLE
Asset versioning UI

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

V1 of asset versioning UI code

### How I Tested These Changes

Manipulated the `asset_reconciliation` example in dagit with the new UI.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #10003** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10003" target="_blank">Graphite</a> 
    * **PR #10005** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10005" target="_blank">Graphite</a> 
      * **PR #9899** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9899" target="_blank">Graphite</a> 
        * **PR #10006** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10006" target="_blank">Graphite</a>  👈
    * **PR #10004** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10004" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10006?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #10003** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10003" target="_blank">Graphite</a> 
    * **PR #10005** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10005" target="_blank">Graphite</a> 
      * **PR #9899** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9899" target="_blank">Graphite</a> 
        * **PR #10006** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10006" target="_blank">Graphite</a>  👈
    * **PR #10004** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10004" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10006?utm_source=stack-comment).

File: js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx [hunk 1]
```diff
@@ -238,7 +249,7 @@ export const AssetGraphExplorerWithData: React.FC<
     // focus on the selected node. (If selection was specified in the URL).
     // Don't animate this change.
     if (lastSelectedNode) {
-      viewportEl.current.zoomToSVGBox(layout.nodes[lastSelectedNode.id].bounds, false);
+      // viewportEl.current.zoomToSVGBox(layout.nodes[lastSelectedNode.id].bounds, false);
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx:252]: Want to rm these comments?

File: js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx [hunk 2]
```diff
@@ -219,14 +225,19 @@ export const AssetGraphExplorerWithData: React.FC<
...
-      onNavigateToForeignNode,
+      onNavigateToSourceAssetNode,
       findAssetLocation,
       lastSelectedNode,
       assetGraphData,
       layout,
     ],
   );
 
+  // const layoutsEqual(layout1: AssetGraphLayout, layout2: AssetGraphLayout) {
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx:236]: Want to rm these comments?

File: js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx [hunk 3]
```diff
@@ -356,9 +368,12 @@ export const AssetGraphExplorerWithData: React.FC<
                         }}
                         style={{overflow: 'visible'}}
                       >
-                        {!graphNode || !graphNode.definition.opNames.length ? (
-                          <ForeignNode assetKey={{path}} />
-                        ) : scale < MINIMAL_SCALE ? (
+                        {false && (!graphNode || !graphNode.definition.opNames.length) ? (
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/asset-graph/AssetGraphExplorer.tsx:371]: I think this isn't quite right but it's fixed in the PR I stacked on this one,  let's leave it for now

File: js_modules/dagit/packages/core/src/assets/LaunchAssetExecutionButton.tsx [hunk 4]
```diff
@@ -425,9 +426,15 @@ export const LAUNCH_ASSET_EXECUTION_ASSET_NODE_FRAGMENT = gql`
     opNames
     jobNames
     graphName
+<<<<<<< HEAD
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/assets/LaunchAssetExecutionButton.tsx:429]: I think these need to be fixed

File: python_modules/dagster/dagster/_core/definitions/asset_group.py [hunk 5]
```diff
@@ -210,6 +210,7 @@ def build_job(
         description = check.opt_str_param(description, "description", "")
         tags = check.opt_mapping_param(tags, "tags", key_type=str)
 
+        print("ASSET GROUP BUILD JOB w SELECTION", selection)
```
Comment (bengotow) [inline python_modules/dagster/dagster/_core/definitions/asset_group.py:213]: rm!

File: python_modules/dagster/dagster/_core/host_representation/repository_location.py [hunk 6]
```diff
@@ -132,12 +132,14 @@ def get_external_pipeline(self, selector: PipelineSelector) -> ExternalPipeline:
         a solid selection is specified, which requires access to the underlying PipelineDefinition
         to generate the subsetted pipeline snapshot."""
         if not selector.solid_selection and not selector.asset_selection:
+            print("NOT GETTING SUBSET")
```
Comment (bengotow) [inline python_modules/dagster/dagster/_core/host_representation/repository_location.py:135]: rm!

File: python_modules/dagster/dagster/_core/host_representation/repository_location.py [hunk 7]
```diff
@@ -132,12 +132,14 @@ def get_external_pipeline(self, selector: PipelineSelector) -> ExternalPipeline:
...
             return self.get_repository(selector.repository_name).get_full_external_job(
                 selector.pipeline_name
             )
 
         repo_handle = self.get_repository(selector.repository_name).handle
 
+        print("GETTING SUBSET")
```
Comment (bengotow) [inline python_modules/dagster/dagster/_core/host_representation/repository_location.py:142]: rm!

**Review by bengotow — APPROVED** [review]
This looks good! I think a few of the foreign asset => source asset verbiage will get broken back out into source asset or "asset link" since that case does still exist, but I've got that incoming in the stacked PR i've been working on. Let's merge this!

**Pull request merged**